src/main/java/jcifs/config/BaseConfiguration.java

```diff
@@ -0,0 +1,822 @@
+                this.resolverOrder.add(ResolverType.RESOLVER_DNS);
+                this.resolverOrder.add(ResolverType.RESOLVER_BCAST);
+            }
+            else {
+                this.resolverOrder.add(ResolverType.RESOLVER_LMHOSTS);
+                this.resolverOrder.add(ResolverType.RESOLVER_DNS);
+                this.resolverOrder.add(ResolverType.RESOLVER_WINS);
+                this.resolverOrder.add(ResolverType.RESOLVER_BCAST);
+            }
+        }
+        else {
+            StringTokenizer st = new StringTokenizer(ro, ",");
+            while ( st.hasMoreTokens() ) {
+                String s = st.nextToken().trim();
+                if ( s.equalsIgnoreCase("LMHOSTS") ) {
+                    this.resolverOrder.add(ResolverType.RESOLVER_LMHOSTS);
+                }
+                else if ( s.equalsIgnoreCase("WINS") ) {
+                    if ( this.winsServer.length == 0 ) {
+                        log.error("UniAddress resolveOrder specifies WINS however " + " WINS server has not been configured");
+                        continue;
+                    }
+                    this.resolverOrder.add(ResolverType.RESOLVER_WINS);
+                }
+                else if ( s.equalsIgnoreCase("BCAST") ) {
+                    this.resolverOrder.add(ResolverType.RESOLVER_BCAST);
+                }
+                else if ( s.equalsIgnoreCase("DNS") ) {
+                    this.resolverOrder.add(ResolverType.RESOLVER_DNS);
+                }
+                else {
+                    log.error("unknown resolver method: " + s);
+                }
+            }
+        }
+    }
+
+
+    protected void initProtocolVersions ( String minStr, String maxStr ) {
+        DialectVersion min = ( minStr != null && !minStr.isEmpty() ) ? DialectVersion.valueOf(minStr) : null;
+        DialectVersion max = ( maxStr != null && !maxStr.isEmpty() ) ? DialectVersion.valueOf(maxStr) : null;
+        initProtocolVersions(min, max);
+    }
+
+
+    protected void initProtocolVersions ( DialectVersion min, DialectVersion max ) {
+        this.minVersion = min != null ? min : DialectVersion.SMB1;
+        this.maxVersion = max != null ? max : DialectVersion.SMB210;
+
+        if ( this.minVersion.atLeast(this.maxVersion) ) {
+            this.maxVersion = this.minVersion;
+        }
+    }
+
+
+    protected void initDisallowCompound ( String prop ) {
+        if ( prop == null ) {
+            return;
+        }
+        Set<String> disallow = new HashSet<>();
+        StringTokenizer st = new StringTokenizer(prop, ",");
+        while ( st.hasMoreTokens() ) {
+            disallow.add(st.nextToken().trim());
+        }
+        this.disallowCompound = disallow;
+    }
+
+
+    protected void initDefaults () throws CIFSException {
+
+        try {
+            "".getBytes(SmbConstants.DEFAULT_OEM_ENCODING);
+        }
+        catch ( UnsupportedEncodingException uee ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)